File: 1695-maximum-erasure-value/1695-maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        
        int score = -1, sum = 0, start = 0;
        vector <int> prefix;
        vector <int> unique (1e4 + 1, -1);
        
        for (int num : nums){
            sum += num;
            prefix.push_back(sum);
        }
        for (int i = 0; i < nums.size(); ++i){
            if (unique[nums[i]] != -1)
                start = max (start, unique[nums[i]] + 1);
            unique[nums[i]] = i;
            int left = (start == 0) ? 0 : prefix[start-1];
            score = max(score, prefix[i] - left);
        }
        return score;
    }
};

File: 1695-maximum-erasure-value/1695-maximum-erasure-value.cs
public class Solution {
    public int MaximumUniqueSubarray(int[] nums) {
        
        int left = 0;
        int currSum = 0;
        int maxSum = 0;
        
        HashSet<int> set = new HashSet<int>();

        for (int right = 0; right < nums.Length; right++)
        {
            while (set.Contains(nums[right]))
            {
                set.Remove(nums[left]);
                currSum -= nums[left];
                left++;
            }
            set.Add(nums[right]);
            currSum += nums[right];
            maxSum = Math.Max(currSum, maxSum);
        }

        return maxSum;
    }
}